yb-voyager/cmd/exportSchema.go

```diff
@@ -167,22 +191,114 @@ func clearSchemaIsExported() {
 	}
 }
 
-func updateIndexesInfoInMetaDB() {
+func updateIndexesInfoInMetaDB() error {
 	log.Infof("updating indexes info in metaDB")
 	if !utils.ContainsString(source.ExportObjectTypeList, "TABLE") {
 		log.Infof("skipping updating indexes info in metaDB since TABLE object type is not being exported")
-		return
+		return nil
 	}
 	indexesInfo := source.DB().GetIndexesInfo()
 	if indexesInfo == nil {
-		return
+		return nil
 	}
 	err := metadb.UpdateJsonObjectInMetaDB(metaDB, metadb.SOURCE_INDEXES_INFO_KEY, func(record *[]utils.IndexInfo) {
 		*record = indexesInfo
 	})
 	if err != nil {
-		utils.ErrExit("update indexes info in meta db: %s", err)
+		return fmt.Errorf("failed to update indexes info in meta db: %w", err)
+	}
+	return nil
+}
+
+func applyMigrationAssessmentRecommendations() error {
+	if skipRecommendations {
+		log.Infof("not apply recommendations due to flag --skip-recommendations=true")
+		return nil
+	}
+
+	assessmentReportPath := lo.Ternary(assessmentReportPath != "", assessmentReportPath,
+		filepath.Join(exportDir, "assessment", "reports", "assessmentReport.json"))
+	if !utils.FileOrFolderExists(assessmentReportPath) {
+		utils.PrintAndLog("migration assessment report file doesn't exists at %q, skipping apply recommendations step...", assessmentReportPath)
+		return nil
+	}
+
+	log.Infof("parsing assessment report json file for applying recommendations")
+	report, err := utils.ParseJsonReportFile[AssessmentReport](assessmentReportPath)
+	if err != nil {
+		return fmt.Errorf("failed to parse json report file %q: %w", assessmentReportPath, err)
+	}
+
+	err = applyColocatedVsShardedTableRecommendation(report.Sharding)
+	if err != nil {
+		return fmt.Errorf("failed to apply colocated vs sharded table recommendation: %w", err)
+	}
+	return nil
+}
+
+func applyColocatedVsShardedTableRecommendation(shardingReport *migassessment.ShardingReport) error {
+	filePath := utils.GetObjectFilePath(schemaDir, "TABLE")
+	if !utils.FileOrFolderExists(filePath) {
+		log.Warnf("required schema file %s does not exists, returning without applying the recommendations", filePath)
+		return nil
+	}
+
+	log.Infof("applying colocated vs sharded table recommendation")
+
+	modifiedFilePath := utils.AddSuffixToFilePath(filePath, "modified")
```

Comment:
It will be better to change the original file name to something like `<existing-file-name>.orig` and then create a new file with the `<existing-file-name>` containing the modified schema. That way, the code after the `export schema` (e.g. analyze-schema doesn't have to worry about two kinds of files. They will continue working as it is.).

Reply:
good idea, this way i don't need to do any changes on the import side.